### PR TITLE
Add the ContentModifier plugin to mitigate security risks automatically and prioritize plugins

### PR DESCRIPTION
**Pull request by AMoo-Miki**
Signed-off-by: Miki <miki@amazon.com>

### Description
* Add the ContentModifier plugin to mitigate security risks automatically
* Add reverse tabnabbing mitigation to ContentModifier
* Add execution priority to the existing plugins
* Prevent unnecessary execution of hooks when the Serve command is used and provide a way to override this behavior
* Optimize hook registration

### Check List
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

**Comment by stockholmux** [comment]
Interesting - I have to ask if this is needed since [tab nabbing has been mitigated at the browser level for quite some time](https://www.zdnet.com/article/chrome-to-block-tab-nabbing-attacks/).

**Comment by AMoo-Miki** [comment]
> Interesting - I have to ask if this is needed since [tab nabbing has been mitigated at the browser level for quite some time](https://www.zdnet.com/article/chrome-to-block-tab-nabbing-attacks/).

I agree that it has been blocked by major browsers for a year now but I also think a year is too short a time to forget about it. I also worry about derivative browsers that might have not pulled in the changes.

**Comment by stockholmux** [comment]
OK - A couple of things:
- This plugin increases a cold build time from avg 9.028 second to 11.231 (+~24%) and subsequent builds by 7.988 seconds to  8.824 (+~10.4%). Build times are critical DX for working on the site.
- This method will continually impact build times as the site grows.
- [IE11 is EoL on June 22, 2022](https://docs.microsoft.com/en-us/lifecycle/faq/internet-explorer-microsoft-edge) (Hooray!), so if we do add this plugin, it should only really be needed until that date.

Another option would be do this through JS only on browsers that don't properly mitigate tabnabbing.

**Comment by AMoo-Miki** [comment]
> OK - A couple of things:
> 
>     * This plugin increases a cold build time from avg 9.028 second to 11.231 (+~24%) and subsequent builds by 7.988 seconds to  8.824 (+~10.4%). Build times are critical DX for working on the site.
> 
>     * This method will continually impact build times as the site grows.
> 
>     * [IE11 is EoL on June 22, 2022](https://docs.microsoft.com/en-us/lifecycle/faq/internet-explorer-microsoft-edge) (Hooray!), so if we do add this plugin, it should only really be needed until that date.
> 
> 
> Another option would be do this through JS only on browsers that don't properly mitigate tabnabbing.

I added mechanism to not run this and the ContentIndexer plugins while developing using `serve`. This should eliminate the burden on contributors.

I did a quick check for the website visitors during the last month and only 1.38% to 1.67% of them were using vulnerable browsers. While those numbers are way too low already, if we use the plugin only for a few months, Android Browser will catch up and IE11 will be EoLed, slashing the vulnerable users down to 0.6%.

File: _plugins/content-modifier.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
```
Comment (stockholmux) [inline _plugins/content-modifier.rb:1]: Does this need a `SPDX-License-Identifier` and a copyright?

File: _plugins/content-modifier.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    90
+  end
+
+  ##
+  # Initializes the singleton by recording the site
+  def self.init(site)
+    @site = site
+
+    # Avoid initializing if serving and not forced to run
+    if site.config["serving"] and (!ENV.key?('JEKYLL_ALLOW_CONTENT_MODIFIER') or ENV['JEKYLL_ALLOW_CONTENT_MODIFIER'] == "false")
```
Comment (stockholmux) [inline _plugins/content-modifier.rb:30]: Can we drop a line or two into README about how to turn it on and off (and what it does)?

**Review by stockholmux — CHANGES_REQUESTED** [review]
A couple of nits and I'm fine with this!

**Comment by stockholmux** [comment]
@AMoo-Miki Any movement on the changes requested?

File: _plugins/content-modifier.rb (hunk 2 above)
Comment (AMoo-Miki) [reply _plugins/content-modifier.rb:34]: Done.

File: _plugins/content-modifier.rb (hunk 1 above)
Comment (AMoo-Miki) [reply _plugins/content-modifier.rb:1]: Added.

**Review by stockholmux — APPROVED** [review]
LGTM

**Pull request merged**